utils/httpx/access.go

```diff
@@ -0,0 +1,46 @@
+package httpx
+
+import (
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+// AccessConfig configures what can be accessed
+type AccessConfig struct {
+	ResolveTimeout time.Duration
+	DisallowedIPs  []net.IP
+}
+
+// NewAccessConfig creates a new access config
+func NewAccessConfig(resolveTimeout time.Duration, disallowedIPs []net.IP) *AccessConfig {
+	return &AccessConfig{
+		ResolveTimeout: resolveTimeout,
+		DisallowedIPs:  disallowedIPs,
+	}
+}
+
+// Allow determines whether the given request should be allowed
+func (c *AccessConfig) Allow(request *http.Request) (bool, error) {
+	host := strings.ToLower(request.URL.Hostname())
+
+	ctx, _ := context.WithTimeout(context.Background(), c.ResolveTimeout)
```

Comment:
Does this work like this? I thought you were meant to save `cancel` then immediately did a `defer cancel()` for contexts?

See: https://developer.squareup.com/blog/always-be-closing/

Reply:
oops.. never used a Context  before. Guess it works til the memory leak gets you